common/lib/xmodule/xmodule/js/src/sequence/display.js

```diff
@@ -0,0 +1,354 @@
+        };
+
+        Sequence.prototype.updatePageTitle = function() {
+            // update the page title to include the current section
+            var positionLink = this.link_for(this.position);
+
+            if (positionLink && positionLink.data('page-title')) {
+                document.title = positionLink.data('page-title') + this.base_page_title;
+            }
+        };
+
+        Sequence.prototype.hookUpContentStateChangeEvent = function() {
+            var self = this;
+
+            return $('.problems-wrapper').bind('contentChanged', function(event, problemId, newContentState, newState) {
+                return self.addToUpdatedProblems(problemId, newContentState, newState);
+            });
+        };
+
+        Sequence.prototype.addToUpdatedProblems = function(problemId, newContentState, newState) {
+            /**
+            * Used to keep updated problem's state temporarily.
+            * params:
+            *   'problem_id' is problem id.
+            *   'new_content_state' is the updated content of the problem.
+            *   'new_state' is the updated state of the problem.
+            * initialize for the current sequence if there isn't any updated problem
+            * for this position.
+            */
+
+            if (!this.anyUpdatedProblems(this.position)) {
+                this.updatedProblems[this.position] = {};
+            }
+
+            // Now, put problem content and score against problem id for current active sequence.
+
+            this.updatedProblems[this.position][problemId] = [newContentState, newState];
+        };
+
+        Sequence.prototype.anyUpdatedProblems = function(position) {
+            /**
+            * check for the updated problems for given sequence position.
+            * params:
+            *   'position' can be any sequence position.
+            */
+            return typeof(this.updatedProblems[position]) !== 'undefined';
+        };
+
+        Sequence.prototype.enableButton = function(buttonClass, buttonAction) {
+            this.$(buttonClass)
+                .removeClass('disabled')
+                .removeAttr('disabled')
+                .click(buttonAction);
+        };
+
+        Sequence.prototype.disableButton = function(buttonClass) {
+            this.$(buttonClass).addClass('disabled').attr('disabled', true);
+        };
+
+        Sequence.prototype.setButtonLabel = function(buttonClass, buttonLabel) {
+            this.$(buttonClass + ' .sr').html(buttonLabel);
+        };
+
+        Sequence.prototype.updateButtonState = function(
+            buttonClass, buttonAction, actionLabelPrefix, isAtBoundary, boundaryUrl
+        ) {
+            var buttonLabel;
+            if (isAtBoundary && boundaryUrl === 'None') {
+                this.disableButton(buttonClass);
+            } else {
+                buttonLabel = actionLabelPrefix + (isAtBoundary ? ' Subsection' : ' Unit');
```

Comment:
Subsection and Unit should be translated-- and placeholders should be used instead of just concatenating onto the actionLabelPrefix.